File: Famicoco/app/src/main/java/com/example/mito/famicoco/TLFragment.java
package com.example.mito.famicoco;

import android.graphics.Bitmap;
import android.os.AsyncTask;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.widget.SwipeRefreshLayout;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.io.InputStream;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.ArrayList;

import butterknife.BindView;
import butterknife.ButterKnife;

import static com.example.mito.famicoco.MainActivity.ServerUrl;
import static com.example.mito.famicoco.MainActivity.tlIcon;

public class TLFragment extends Fragment implements SwipeRefreshLayout.OnRefreshListener {      //タイムライン用のフラグメント

    private CustomAdapter adapter;
    private ArrayList<CustomData> list;

    @BindView(R.id.list)
    ListView listView;
    @BindView(R.id.swipe_tl)
    SwipeRefreshLayout swipeRefreshLayout;

    @Override
    public void onCreate(Bundle saveInterfaceState) {
        super.onCreate(saveInterfaceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_tl, container, false);
        ButterKnife.bind(this, v);

        swipeRefreshLayout.setColorSchemeResources(R.color.MainColorDark,
                R.color.MainColor, R.color.MainColorDark, R.color.MainColor);
        swipeRefreshLayout.setOnRefreshListener(this);
        return v;
    }

    @Override
    public void onResume() {
        super.onResume();
        UpData();
    }

    @Override
    public void onActivityCreated(Bundle bundle) {
        super.onActivityCreated(bundle);
        setHasOptionsMenu(true);

        list = new ArrayList<>();
        adapter = new CustomAdapter(getContext(), 0, list);
    }

    public TLFragment() {
    }

    public void UpData() {
        AsyncTask<URL, Void, ArrayList<TLUpdateItem>> task = new AsyncTask<URL, Void, ArrayList<TLUpdateItem>>() {
            @Override
            protected ArrayList<TLUpdateItem> doInBackground(URL... values) {
                HttpURLConnection con = null;
                TLUpdateItem item;
                ArrayList<TLUpdateItem> tlUpdateItems = new ArrayList<>();
                try {
                    // アクセス先URL
                    final URL url = values[0];
                    // ローカル処理
                    // コネクション取得
                    con = (HttpURLConnection) url.openConnection();
                    con.setDoInput(true);
                    con.connect();
                    InputStream in = con.getInputStream();
                    String readSt = HttpGetTask.readInputStream(in);
                    Log.d("readst", readSt);

                    //// 配列を取得する場合
                    JSONArray jsonArray = new JSONArray(readSt);

                    int m = jsonArray.length();
                    for (int i = 0; i < m; i++) {
                        JSONObject data = jsonArray.getJSONObject(i);
                        item = new TLUpdateItem(data);
                        tlUpdateItems.add(item);
                    }
                } catch (JSONException | IOException e) {
                    e.printStackTrace();
                } finally {
                    if (con != null) {
                        // コネクションを切断
                        con.disconnect();
                    }
                }
                return tlUpdateItems;
            }

            @Override
            protected void onPostExecute(ArrayList<TLUpdateItem> tlUpdateItemArrayList) {
                list.clear();
                adapter = new CustomAdapter(getContext(), 0, list);
                int m = tlUpdateItemArrayList.size();
                for (int i = 0; i < m; i++) {
                    TLUpdateItem tlUpdateItem = tlUpdateItemArrayList.get(i);
                    list.add(0, tlUpdateItem.toCustomData());
                    listView.setAdapter(adapter);
                }
                Log.d("hoge", "here");
            }
        };

        try {
            URL url = new URL(ServerUrl + "get_tl/");
            task.execute(url);
        } catch (MalformedURLException e) {
            e.printStackTrace();
        }
    }

    @Override
    public void onRefresh() {
        Log.d("Refresh", "screen");
        UpData();
        swipeRefreshLayout.setRefreshing(false);
    }
}

class TLUpdateItem {
    private String text;
    private String time;
    private Bitmap icon;

    TLUpdateItem(JSONObject object) throws JSONException {
        this.time = object.getString("time");
        this.text = object.getString("action");
        String type = object.getString("action_num");
        this.icon = tlIcon[type.charAt(0) - '0'];
    }

    CustomData toCustomData() {
        CustomData customData = new CustomData();
        customData.setComment(this.text);
        customData.setTime(this.time);
        customData.setIcon(this.icon);
        return customData;
    }
}

File: Famicoco/app/src/main/java/com/example/mito/famicoco/TLFragment.kt
package com.example.mito.famicoco

import android.annotation.SuppressLint
import android.graphics.Bitmap
import android.os.AsyncTask
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ListView
import androidx.fragment.app.Fragment
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
import butterknife.BindView
import butterknife.ButterKnife
import com.example.mito.famicoco.MainActivity.Companion.ServerUrl
import com.example.mito.famicoco.MainActivity.Companion.tlIcon
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL
import java.util.ArrayList

class TLFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener { // タイムライン用のフラグメント
    
    private lateinit var adapter: CustomAdapter
    private lateinit var list: ArrayList<CustomData>
    
    @BindView(R.id.list)
    internal lateinit var listView: ListView
    @BindView(R.id.swipe_tl)
    internal lateinit var swipeRefreshLayout: SwipeRefreshLayout
    
    override fun onCreateView(
            inflater: LayoutInflater,
            container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(
                R.layout.fragment_tl,
                container,
                false
        )
        ButterKnife.bind(
                this,
                view
        )
        
        swipeRefreshLayout.setColorSchemeResources(
                R.color.MainColorDark,
                R.color.MainColor,
                R.color.MainColorDark,
                R.color.MainColor
        )
        swipeRefreshLayout.setOnRefreshListener(this)
        return view
    }
    
    override fun onResume() {
        super.onResume()
        update()
    }
    
    override fun onActivityCreated(bundle: Bundle?) {
        super.onActivityCreated(bundle)
        setHasOptionsMenu(true)
        
        list = ArrayList()
        adapter = CustomAdapter(
                context,
                0,
                list
        )
    }
    
    private fun update() {
        @SuppressLint("StaticFieldLeak")
        val task = object : AsyncTask<URL, Void, ArrayList<TLUpdateItem>>() {
            override fun doInBackground(vararg values: URL): ArrayList<TLUpdateItem> {
                var con: HttpURLConnection? = null
                var item: TLUpdateItem
                val tlUpdateItems = ArrayList<TLUpdateItem>()
                try {
                    // アクセス先URL
                    val url = values[0]
                    // ローカル処理
                    // コネクション取得
                    con = url.openConnection() as HttpURLConnection
                    con.doInput = true
                    con.connect()
                    val `in` = con.inputStream
                    val readSt = HttpGetTask.readInputStream(`in`)
                    
                    // // 配列を取得する場合
                    val jsonArray = JSONArray(readSt)
                    
                    val m = jsonArray.length()
                    for (i in 0 until m) {
                        val data = jsonArray.getJSONObject(i)
                        item = TLUpdateItem(data)
                        tlUpdateItems.add(item)
                    }
                } catch (e: Exception) {
                    e.printStackTrace()
                } finally {
                    con?.disconnect()
                }
                return tlUpdateItems
            }
            
            override fun onPostExecute(tlUpdateItemArrayList: ArrayList<TLUpdateItem>) {
                list.clear()
                adapter = CustomAdapter(
                        context,
                        0,
                        list
                )
                val m = tlUpdateItemArrayList.size
                for (i in 0 until m) {
                    val tlUpdateItem = tlUpdateItemArrayList[i]
                    list.add(
                            0,
                            tlUpdateItem.toCustomData()
                    )
                    listView.adapter = adapter
                }
            }
        }
        
        try {
            val url = URL(ServerUrl + "get_tl/")
            task.execute(url)
        } catch (e: MalformedURLException) {
            e.printStackTrace()
        }
    }
    
    override fun onRefresh() {
        update()
        swipeRefreshLayout.isRefreshing = false
    }
}

internal class TLUpdateItem @Throws(JSONException::class)
constructor(`object`: JSONObject) {
    private val text: String = `object`.getString("action")
    private val time: String = `object`.getString("time")
    private val icon: Bitmap
    
    init {
        val type = `object`.getString("action_num")
        this.icon = tlIcon[type[0] - '0']
    }
    
    fun toCustomData(): CustomData {
        val customData = CustomData()
        customData.comment = this.text
        customData.timeNow = this.time
        customData.icon = this.icon
        return customData
    }
}